ru/lessons/libraries/guardian.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: page
+title: Guardian (Основы)
+category: libraries
+order: 1
+lang: ru
+---
+
+[Guardian](https://github.com/ueberauth/guardian) &mdash; часто используемая библиотека, основанная на [JWT](https://jwt.io/) (Javascript Web Token).
+
+{% include toc.html %}
+
+## JWT
+
+Что такое JWT? JWT занимается созданием токена для аутентификации. В то время как многие системы аутентификации предоставляют доступ только к идентификатору ресурса, JWT позволяет хранить еще и другую информацию, такую как:
+
+* Кем выдан токен
+* Для кого выдан токен
+* Какие системы должны использовать этот токен
+* На какое время он выдан
+* Каков тип токена
+* Какие права есть у владельца данного токена
+
+Это всё &mdash; простые примеры полей в JWT. Технология позволяет добавлять любую дополнительную информацию, которой требует логика приложения. Главное &mdash; не забывайте что этот токен должен быть коротким, ведь он должен поместиться в HTTP заголовке.
+
+Следствием такого подхода является тот факт, что JWT могут передаваться как полноценный целостный элемент данных идентификации пользователя.
+
+### Когда их использовать
+
+Токены JWT могут использваться для аутентификации в любой части приложения:
+
+* Одностраничные приложения
+* Контроллеры (как с помощью браузерной сессии, так и api: с помощью HTTP заголовков)
+* Phoenix Channels
+* Запросы между сервисами
+* Запросы между процессами
+* Функциональность "запомнить меня"
```

Comment:
In the original lesson `3rd Party access (OAuth)` and `Other interfaces - raw TCP, UDP, CLI` are also mentioned in this list.